### PR TITLE
Replace C++ GoogleTest tests with JS tests using Mocha and Chai

### PR DESCRIPTION
**Pull request by darraghjburke**
Removes GoogleTest dependency and C++ tests for UrlLib. In their place, adds basic Javascript tests for XMLHTTPRequest using Mocha.js and Chai.js.

Includes several hacks that should be cleaned up, marked with TODOs.

Creates base framework for #802.

### CODE REVIEW

**Review by bghgary — APPROVED** [review]
Looks okay overall. Obviously lots of TODOs still, but that's okay with me.

File: Apps/UnitTests/Scripts/tests.js [hunk 1]
```diff
@@ -31,21 +31,21 @@ const expect = chai.expect;
...
-        const xhr = await makeXHR("GET", "https://babylonjs.com/invalid");
+        const xhr = await createRequest("GET", "https://babylonjs.com/invalid");
         expect(xhr.status).to.equal(404);
     })
 
 });
 
 
-mocha.run(function (failures) {
+mocha.run( failures => {
```
Comment (bghgary) [inline Apps/UnitTests/Scripts/tests.js:49]: nit: there typically isn't a space here.  might consider auto formatting the whole file.
```suggestion
mocha.run(failures => {
```

Comment (darraghjburke) [reply Apps/UnitTests/Scripts/tests.js]: fixed! thanks! would it be possible to add a linting step to the CI checks?

Comment (bghgary) [reply Apps/UnitTests/Scripts/tests.js]: We could. We probably should transition this to a TS file before doing that though.

Comment (darraghjburke) [reply Apps/UnitTests/Scripts/tests.js]: What should we use to compile TS files? As far as I know we are not using TS anywhere in the Babylon Native repo yet

Comment (bghgary) [reply Apps/UnitTests/Scripts/tests.js]: I'm not sure yet. Something similar to how Babylon.js does it probably.

**Pull request merged**